history/src/main/scala/org/apache/spark/deploy/history/SimpleFsHistoryProvider.scala

```diff
@@ -0,0 +1,428 @@
+
+  private[history] def initialize(): Unit = {}
+  
+  override def getEventLogsUnderProcess(): Int = 0
+
+  override def getLastUpdatedTime(): Long = 0
+  
+  override def getListing(): Iterator[ApplicationInfo] = { null }
+  
+  override def onUIDetached(appId: String, attemptId: Option[String], ui: SparkUI): Unit = { }
+   
+  private val activeUIs = new mutable.HashMap[(String, Option[String]), LoadedAppUI]()
+  
+  def getApplicationAttemptInfo(fileStatus: FileStatus): Option[ApplicationAttemptInfo] = {
+ 
+    // When getting the Application Info, we really only want the key details - Name, ID, start time, stop time
+    val eventsFilter: ReplayEventsFilter = { eventString =>
+      eventString.startsWith(APPL_START_EVENT_PREFIX) ||
+      eventString.startsWith(APPL_END_EVENT_PREFIX)
+    }
+
+    // need to get the ApplicationAttemptInfo from the replay to return
+    val logPath = fileStatus.getPath()
+    val bus = new ReplayListenerBus()
+    val listener = new AppListingListener(fileStatus, clock)
+    bus.addListener(listener)
+    replay(fileStatus, bus, eventsFilter)
+
+    listener.applicationAttemptInfo
+  }
+  
+  def getApplicationInfoWrapper(appId: String): Option[ApplicationInfoWrapper] = {
+    val fileStatus = fs.getFileStatus(new Path(logDir, appId))
+    val logPath = fileStatus.getPath()
+    val bus = new ReplayListenerBus()
+    val listener = new AppListingListener(fileStatus, clock)
+    bus.addListener(listener)
+    replay(fileStatus, bus, SELECT_ALL_FILTER)
+    listener.applicationInfo
+  }
+
+  override def getApplicationInfo(appId: String): Option[ApplicationInfo] = {
+    getApplicationInfoWrapper(appId).map(app => {app.info})
+  }
+
+  override def getAppUI(appId: String, attemptId: Option[String]): Option[LoadedAppUI] = {
+    getApplicationInfoWrapper(appId).map(applicationInfo => {
+      val attemptWrapper = applicationInfo.attempts.head
+
+      val conf = this.conf.clone()
+      val secManager = new SecurityManager(conf)
+      secManager.setAcls(HISTORY_UI_ACLS_ENABLE)
+      secManager.setAdminAcls(attemptWrapper.adminAcls.getOrElse(""))
+      secManager.setViewAcls(applicationInfo.attempts.head.info.sparkUser, attemptWrapper.adminAclsGroups.getOrElse(""))
+      secManager.setAdminAclsGroups(attemptWrapper.adminAclsGroups.getOrElse(""))
+      secManager.setViewAclsGroups(attemptWrapper.viewAclsGroups.getOrElse(""))
+      
+      val kvstore = createInMemoryStore(attemptWrapper)
+      val something = attemptWrapper.info.appSparkVersion
+      val ui = SparkUI.create(None, new AppStatusStore(kvstore), conf, secManager, applicationInfo.info.name,
+        HistoryServer.getAttemptURI(applicationInfo.info.id, attemptWrapper.info.attemptId),
+        attemptWrapper.info.startTime.getTime(),
+        attemptWrapper.info.appSparkVersion
+      )
+
+    val loadedUI = LoadedAppUI(ui)
+
+    synchronized {
+      activeUIs((appId, attemptId)) = loadedUI
```

Comment:
We don't reference this activeUIs map anywhere else either, it can likely be dropped